File: scrimage-core/src/test/scala/com/sksamuel/scrimage/nio/PngReaderTest.scala
package com.sksamuel.scrimage.nio

import java.io.File
import java.nio.file.Files

import org.scalatest.{ Matchers, WordSpec }

class PngReaderTest extends WordSpec with Matchers {

  "PngReader" should {
    "be able to read pngs of all channels" in {
      val files = new File(getClass.getResource("/png").getFile).listFiles()
      val images = files map { file => PngReader.read(Files.readAllBytes(file.toPath)).get }
      images.length shouldBe 13
    }
  }
}

File: scrimage-core/src/test/kotlin/com/sksamuel/scrimage/core/nio/PngReaderTest.kt
@file:Suppress("BlockingMethodInNonBlockingContext")

package com.sksamuel.scrimage.core.nio

import com.sksamuel.scrimage.nio.PngReader
import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.shouldBe
import java.io.File
import java.nio.file.Files

class PngReaderTest : WordSpec({

   "PngReader" should {
      "be able to read pngs of all channels"  {
         val files = File(javaClass.getResource("/png").file).listFiles()
         val images = files.map { file -> PngReader().read(Files.newInputStream(file.toPath()), null) }
         images.size shouldBe 13
      }
   }

})
